core/src/price_estimation/average_price_source.rs

```diff
@@ -56,10 +56,22 @@ fn lossless_merge<T: Eq + Hash, U>(map_collection: Vec<HashMap<T, U>>) -> HashMa
 
 fn average_prices(price_maps: Vec<HashMap<TokenId, u128>>) -> HashMap<TokenId, u128> {
     // Lossless merger of the collection of hash maps. That is, putting all
-    // available prices for each token into a list to be averaged at the end.
+    // available prices for each token into a list (ignorign 0 values) to be
+    // averaged at the end.
     lossless_merge(price_maps)
         .iter()
-        .map(|(token, prices)| (*token, prices.iter().sum::<u128>() / prices.len() as u128))
+        .filter_map(|(token, prices)| {
+            let filtered_prices: Vec<u128> =
+                prices.iter().filter(|&&p| p > 0u128).cloned().collect();
```

Comment:
ubernit: I would slightly prefer not collecting here if possible (I think it should be with `fold`). However, if it makes the code harder to read, then we should leave it as is.